apps/src/lib/ui/accounts/DeleteAccount.jsx

```diff
@@ -191,7 +192,7 @@ export default class DeleteAccount extends React.Component {
         <DeleteAccountDialog
           isOpen={isDeleteAccountDialogOpen}
           isPasswordRequired={isPasswordRequired}
-          isTeacher={isTeacher}
+          isTeacher={isTeacher && hasStudents}
```

Comment:
Perhaps this property should be changed from `isTeacher` to something more specific, like `warnAboutDeletingStudents` to avoid confusion in the future.